### ISSUE TITLE
Rule suggestion: only throw Error and subclasses

### ISSUE THREAD

**Issue by danvk** [palantir/tslint#1465]
I'd like a rule to flag this as invalid:

``` typescript
throw "An error occurred.";
```

in favor of:

``` typescript
throw new Error("An error occurred.");
```

There are good reasons to throw Errors, rather than strings. See http://www.devthought.com/2011/12/22/a-string-is-not-an-error/

**Comment by adidahiya** [comment]
sounds good, we could have `no-throw-literal` like eslint: http://eslint.org/docs/rules/no-throw-literal

**Comment by JoshuaKGoldberg** [comment]
Perfect use case for TypeScript over raw JavaScript... With the type checker this should probably be something like `only-throw-errors`.

```typescript
class Foo { }
const foo = new Foo();
throw foo(); // should be an error
```

> Due to the limits of static analysis, this rule cannot guarantee that you will only throw Error objects.

**Comment by andy-hanson** [comment]
Should be fixed by #2073.

**Comment by adidahiya** [comment]
well, #2073 did not merge, but #1878 (`no-string-throw`) did
